sklearn/preprocessing/_function_transformer.py

```diff
@@ -241,29 +260,53 @@ def transform(self, X):
         """
         X = self._check_input(X, reset=False)
         out = self._transform(X, func=self.func, kw_args=self.kw_args)
+        output_config = _get_output_config("transform", self)["dense"]
 
         if hasattr(out, "columns") and self.feature_names_out is not None:
```

Comment:
not to change in this PR, but I feel uncomfortable with ducktyping `columns` here instead of checking against our adapters. Thinking of xarray for instance here.